### DISCUSSION TITLE
Client token error connecting to AWS DynamoDB in production

### DISCUSSION THREAD

**Discussion by asfand987** [vercel/next.js#64263] [Help]
### Summary

Hello, My website is deployed to Vercel and requires access to DynamoDB. Up until today did not have any issues but not all of a sudden I am unable to access the resource anymore. 

I have tried, creating new key, new user, new role. I have set the user variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY as environmental variables in Vercel but still can't get access. The permission AmazonDynamoDBFullAccess is also set. In local i have no issues and its working fine. It was working on vercel fine for weeks/months too. I am not sure what has happened.

Help would be appreciated.

### Additional information

```js
I get the following error

`{"name":"UnrecognizedClientException","$fault":"client","$metadata":{"httpStatusCode":400,"requestId":"8H7RLVD51CC2UTI845CVASRGGNVV4KQNSO5AEMVJF66Q9ASUAAJG","attempts":1,"totalRetryDelay":0},"__type":"com.amazon.coral.service#UnrecognizedClientException","message":"The security token included in the request is invalid"}`

This is my client
`const client = new DynamoDBClient({ region: REGION, credentials: fromEnv() });`
```

**Comment by asfand987** [comment]
My enviromental variables set in Vercel 

```
AWS_SECRET_ACCESS_KEY
Development, Preview, Production

•••••••••••••••
Updated 10m ago
AWS_ACCESS_KEY_ID
Development, Preview, Production

•••••••••••••••
Updated 11m ago
AWS_DYNAMODB_TABLE_NAME
Development, Preview, Production

•••••••••••••••
Added 49d ago
AWS_DYNAMODB_TABLE_KEY
Development, Preview, Production

•••••••••••••••
Added 49d ago
AWS_REGION
Development, Preview, Production
```

**Reply by h0i5** [reply]
Hey! Did you manage to find a fix? 
Thanks !!

**Reply by massfords** [reply]
There's an interesting note in the Vercel [docs](https://vercel.com/docs/environment-variables/reserved-environment-variables#allowed-environment-variables) about what AWS environment variables are allowed. 

See the paragraph below what's listed as allowed:

>These variables may appear in your Vercel Functions even if you don't set them in your project explicitly. These values do not grant any AWS permissions and are not usable as AWS credentials. Configure your own AWS credentials using environment variables or set up

I had an app working for 2 years and then suddenly stopped working with credentials errors. The culprit was the process.env having a value for `AWS_SESSION_TOKEN` which broke the auth. The solution was to explicitly configure the credentials for each aws-sdk v3 client as opposed to using helpers like `fromEnv()` or the default provider chain.

**Reply by wildky** [reply]
Wow I thought I was going crazy, this just happened to me as well.

Thanks @massfords for commenting here - I came to this thread last night before you added a comment and was very disheartened 😆. It is very relieving to know I am not the only one! I will try what you recommend.

**Reply by homit-dalia** [reply]
My project worked fine for a year, and then it suddenly started throwing this error. The issue goes away if I manually pass credentials to every client instead of relying on the library to automatically pick them up from the environment, similar to what @massfords suggested.

For example, replace

```
const client = new DynamoDBClient({
  region: process.env.AWS_REGION,
})
```

to

```
const client = new DynamoDBClient({
  region: process.env.AWS_REGION,
  credentials: {
    accessKeyId: process.env.AWS_ACCESS_KEY_ID_NEW,
    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY_NEW,
  },
})
```
